apps/passport-client/components/modals/AccountImportModal.tsx

```diff
@@ -0,0 +1,187 @@
+import { Spacer } from "@pcd/passport-ui";
+import { PCD } from "@pcd/pcd-types";
+import { useCallback, useEffect, useState } from "react";
+import styled from "styled-components";
+import { useFilePicker } from "use-file-picker";
+import { useDispatch, usePCDCollection } from "../../src/appHooks";
+import { Button } from "../core";
+
+type ImportState =
+  | {
+      valid: true;
+      imported: false;
+      pcds: PCD[];
+      folders: FolderMap;
+      conflicts: number;
+    }
+  | { valid: true; imported: true; added: number }
+  | { valid: false };
+
+type FolderMap = { [pcdId: string]: string };
+
+export function AccountImportModal() {
+  const { openFilePicker, filesContent } = useFilePicker({
+    accept: ".json",
+    multiple: false
+  });
+
+  const pcdCollection = usePCDCollection();
+  const dispatch = useDispatch();
+
+  const [importState, setImportState] = useState<ImportState | undefined>();
+
+  const importPCDs = useCallback(() => {
+    // Should never happen, but makes TypeScript happy that we checked for it
+    if (!importState.valid || importState.imported === true) return;
+
+    const addedPCDs: PCD[] = [];
+    for (const pcd of importState.pcds) {
+      if (!pcdCollection.hasPCDWithId(pcd.id)) {
+        try {
+          pcdCollection.add(pcd);
```

Comment:
This works fine for import.  Given I recently realized that PCD ordering is a user-visible piece of state, let me think through that here.  This logic doesn't attempt to preserve PCD ordering but puts imported ones at the end of the folder order as if they were new.  Doing otherwise would be complicated, so I'm not recommending it, just pointing it out.